app/models/katello/host_collection.rb

```diff
@@ -24,20 +24,33 @@ class HostCollection < Katello::Model
                                                      :message => N_("must be a positive integer value.")}
 
     alias_attribute :content_host_limit, :max_content_hosts
-    validate :validate_max_content_hosts
+
+    validate :max_content_hosts_check, :if => ->(host_collection) { host_collection.new_record? || host_collection.max_content_hosts_changed? }
+
+    validate :max_content_host_value_exists
+
+    validate :max_content_hosts_not_exceeded, :on => :create
 
     scoped_search :on => :name, :complete_value => true
     scoped_search :on => :organization_id, :complete_value => true
 
-    def validate_max_content_hosts
-      if new_record? || max_content_hosts_changed?
-        if (!unlimited_content_hosts) && (systems.length > 0 && (systems.length > max_content_hosts))
-          errors.add :content_host_limit, _("may not be less than the number of content hosts associated with the host collection.")
-        elsif (max_content_hosts == 0)
-          errors.add :content_host_limit, _("may not be set to 0.")
-        elsif (unlimited_content_hosts == false) && (max_content_hosts.nil?)
-          errors.add :max_content_hosts, _("must be given a value if this host collection is not unlimited.")
-        end
+    def max_content_hosts_check
+      if (!unlimited_content_hosts) && (systems.length > 0 && (systems.length.to_i > max_content_hosts.to_i)) && max_content_hosts_changed?
```

Comment:
Also, I think you can drop the `max_content_hosts_changed?` check here since it's now on the `validate`